premailer/premailer.py

```diff
@@ -417,10 +417,18 @@ def transform(self, pretty_print=True, **kwargs):
                         parent.attrib[attr].startswith('cid:')
                     ):
                         continue
-                    parent.attrib[attr] = urljoin(
+                    url = parent.attrib[attr]
+
+                    if not url.startswith("//"):
+                        url = url.lstrip('/')
```

Comment:
I struggle to parse this. So, if `url` is something like `/path/to.css` you're going to change that to `path/to.css`. That is not necessary because `urljoin()` is really clever:
```python
>>> urljoin('https://www.peterbe.com/', '/path/to.css')
'https://www.peterbe.com/path/to.css'
>>> urljoin('https://www.peterbe.com/', 'path/to.css')
'https://www.peterbe.com/path/to.css'
>>> urljoin('https://www.peterbe.com', '/path/to.css')
'https://www.peterbe.com/path/to.css'
>>> urljoin('https://www.peterbe.com', 'path/to.css')
'https://www.peterbe.com/path/to.css'
```

Best of all, `urljoin()` understands `//` URLs too. E.g. 
```
>>> urljoin('https://www.peterbe.com', '//domain.com/path/to.css')
'https://domain.com/path/to.css'
>>> urljoin('http://www.peterbe.com', '//domain.com/path/to.css')
'http://domain.com/path/to.css'
```
which is ideal.